dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/translators/documentables/DefaultPageCreator.kt

```diff
@@ -643,128 +671,180 @@ public open class DefaultPageCreator(
             }
         }
 
-    private fun DocumentableContentBuilder.functionsBlock(
-        name: String,
-        list: Collection<DFunction>
-    ) {
+    private fun DocumentableContentBuilder.typesBlock(types: List<Documentable>) {
+        if (types.isEmpty()) return
+
+        val grouped = types
+            // This groupBy should probably use LocationProvider
+            .groupBy(Documentable::name)
+            .mapValues { (_, elements) ->
+                // This hacks displaying actual typealias signatures along classlike ones
+                if (elements.any { it is DClasslike }) elements.filter { it !is DTypeAlias } else elements
+            }
+
+        val groups = grouped.entries
+            .sortedWith(compareBy(nullsFirst(canonicalAlphabeticalOrder)) { it.key })
+            .map { (name, elements) ->
+                DivergentElementGroup(
+                    name = name,
+                    kind = ContentKind.Classlikes,
+                    contentTypeOverride = null,
+                    elements = elements
+                )
+            }
+
         divergentBlock(
-            name,
-            list.sorted(),
-            ContentKind.Functions,
-            extra = mainExtra
+            name = "Types",
+            kind = ContentKind.Classlikes,
+            extra = mainExtra,
+            contentType = BasicTabbedContentType.TYPE,
+            groups = groups
         )
     }
 
-    private fun DocumentableContentBuilder.propertiesBlock(
+    private fun DocumentableContentBuilder.declarationsBlock(
         name: String,
-        list: Collection<DProperty>
+        isFunctions: Boolean,
+        declarations: List<Documentable>,
```

Comment:
> it may be better/easier (or not) to split extensions and regular callables inside this function

This is specifically the cause, why I've had to refactor `divergentBlock` and to path extensions externally, as we can't distinguish it here.

> A function with a very broad name declarationsBlock() with a param isFunctions: Boolean that does else -> Properties - it's asking for trouble

Yeah, I agree, though, it's hard to find a good name for a function which should work for member/extensions functions/properties and not other entities, while having possibility to customise there handling.
Let's check our variants:
1. pass `ContentKind` - this sounds even better than boolean flag for me, as in this case, in future, someone will be able to easily add one more condition to `when` and so one more time make this function a responsibility monster — which should have not, as handling of types (class likes) most likely will always be different comparing to properties and functions, which are rather similar in handling
2. Deducing the type from `declarations` on the other hand could be nice, but it can suffer from the same issue above.
3. Leave `isFunctions` boolean flag but rename function to something like `functionsOrPropertiesBlock` + for clarity introduce 2 functions near for just functions and properties

I'm more in favour of variant 3 to make behaviour more explicit, that it's applied only for functions and properties.

WDYT?